### PR TITLE
Adding CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by Maxusmusti**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Contributing to the CodeFlare Operator
+
+Here are a few things to go over before getting started with CodeFlare Operator development:
+
+## Environment setup
+
+The following should be installed in your working environment:
+installation/)
```
Comment (MichaelClifford) [inline CONTRIBUTING.md:8]: is this a typo?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Under the `api` dir, the MCAD and InstaScale custom resources are defined:
+ - See `mcad_types.go` and `instascale_types.go`
+
+Under `config/internal` are where the MCAD and InstaScale resource templates can be found:
+ - Sorted under `mcad` and `insascale` subdirs
+
+The code for MCAD/InstaScale resource reconsilliation can be found in the `controllers` dir:
+ - See `mcad_controller.go` and `instascale_controller.go`
+
+The main entrypoint for the operator is `main.go`
```
Comment (MichaelClifford) [inline CONTRIBUTING.md:25]: Might want to move this to the top of this section?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ - `make image-push -e IMG<image-repo/image-name>`
+
+For deploying onto a cluster:
+ - First, either set `KUBECONFIG` or ensure you are logged into a cluster in your environment
+ - `make install`
+ - `make deploy -e IMG=<image-repo/image-name>`
+
+## Testing
+The CodeFlare Operator currently has unit tests and pre-commit checks
+ - To enable and view pre-commit checks: `pre-commit install`
```
Comment (anishasthana) [inline CONTRIBUTING.md:45]: I'd add a note here basically saying this is required, since we do have a CI check for it.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ - GCC
+
+## Basic Overview
+Under the `api` dir, the MCAD and InstaScale custom resources are defined:
+ - See `mcad_types.go` and `instascale_types.go`
+
+Under `config/internal` are where the MCAD and InstaScale resource templates can be found:
+ - Sorted under `mcad` and `insascale` subdirs
+
+The code for MCAD/InstaScale resource reconsilliation can be found in the `controllers` dir:
```
Comment (anishasthana) [inline CONTRIBUTING.md:22]:
```suggestion
The code for MCAD/InstaScale resource reconcilliation can be found in the `controllers` dir:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+The main entrypoint for the operator is `main.go`
+
+## Building and Deployment
+If changes are made in the `api` dir, run: `make manifests`
+ - This will generate new CRDs and associated files
+
+If changes are made to any Go code (like in the `controllers` dir for example), run: `make`
+ - This will check and build/compile the modified code
+
+For building and pushing a new version of the operator image:
```
Comment (MichaelClifford) [inline CONTRIBUTING.md:34]: Is this something you want developers to do? Is the intention that they would be pushed to their own dev image repos?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+   - [Download release](https://go.dev/dl/)
+   - [Install Instructions](https://go.dev/doc/install)
+ - [Operator SDK](https://sdk.operatorframework.io/docs/installation/)
+ - GCC
+
+## Basic Overview
+Under the `api` dir, the MCAD and InstaScale custom resources are defined:
+ - See `mcad_types.go` and `instascale_types.go`
+
+Under `config/internal` are where the MCAD and InstaScale resource templates can be found:
```
Comment (KPostOffice) [inline CONTRIBUTING.md:19]:
```suggestion
The MCAD and InstaScale resource templates can be found under `config/internal`:
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+The following should be installed in your working environment:
+installation/)
+ - Go 1.18.X
+   - [Download release](https://go.dev/dl/)
+   - [Install Instructions](https://go.dev/doc/install)
+ - [Operator SDK](https://sdk.operatorframework.io/docs/installation/)
+ - GCC
+
+## Basic Overview
+Under the `api` dir, the MCAD and InstaScale custom resources are defined:
```
Comment (KPostOffice) [inline CONTRIBUTING.md:16]:
```suggestion
The MCAD and InstaScale custom resources are defined under the `api` dir:
```

File: CONTRIBUTING.md (hunk 5 above)
Comment (anishasthana) [reply CONTRIBUTING.md:34]: I think in this case it's something developers would need to do for their own repos

**Review by MichaelClifford — APPROVED** [review]
LGTM

I'll be attempting to actually use it today setting up my dev environment. I'll make edits or add issues as they arise.

**Pull request merged**